### DISCUSSION TITLE
Writing DAX Query with filter for Report Builder List

### DISCUSSION THREAD

**Discussion by patrickbcox** [DaxStudio/DaxStudio#778] [Q&A] [answered]
I have a single fact table and a date table in power BI.  I pulled those into Report Builder and now I need to write a query to create a list of years that I can assign to a parameter that report users can use to select the year they want to report on.  the data table has more years in it that data that exists in the fact table.  So I would like to write a DAX statement to simply display only the years that have records in the fact table.  Any suggestions?  When I write the following, I get all years in the data table.  how can I filter this list based on the fact table?  Thanks!

EVALUATE
SUMMARIZECOLUMNS( DimDate[Year])

**Comment by dgosbell** [comment, marked as answer]
I don't know if you've seen the articles and videos talking about how filters in the CALCULATE function are actually tables - but the filters in the SUMMARIZECOLUMNS function work the same way. So you can do the following:

EVALUATE
SUMMARIZECOLUMNS( DimDate[Year], 'yourFactTable' )

and you should get a list of years that are filtered to just those in 'yourTactTable' (obviously you need to put in the actually name of your fact table)

**Reply by patrickbcox** [reply]
Wow, that was easy!  Thanks!  Can you point me somewhere to get the basics of writing DAX queries?  I am pretty comfortable with creating measures in Power BI but DAX queries are a bit different.  Thanks again!

**Reply by dgosbell** [reply]
The only thing I am aware of is the tutorial I wrote on the DAX Studio site https://daxstudio.org/tutorials/writing-dax-queries/

**Reply by marcosqlbi** [reply]
We have an introductive section about Writing queries in DAX in The Definitive Guide to DAX and in our Mastering DAX video course. I am also working on a more specific training about that, but it will be ready in the first half of 2022. 
https://www.sqlbi.com/books/the-definitive-guide-to-dax-2nd-edition/
https://www.sqlbi.com/p/mastering-dax-video-course/

**Reply by patrickbcox** [reply]
Thanks Marco!  I have the video course but have not gotten to that section so I will watch it now.  Since I am now working in Report Builder I need to write some queries there so this will be very helpful.  And I look forward to your new training next year!
